en/animation/skeletal-animation.md

```diff
@@ -55,11 +55,28 @@ It is recommended that projects with high quality skinning animations try to ena
 
 ## Socket System
 
-If you need to attach some external nodes to a given skeletal joint, you need to use the **Socket System** of the skeletal animation component.
-
-- Create a new child node under the skeletal animation component to be docked (the immediate parent node should be the node where the animation component is located).
-- Add an array element to the `sockets` property of the skeletal animation component, select the `path` of the skeleton to be attached from the drop-down list (note that the defaultClip of the skeletal animation component must have a value, the options in the drop-down list depend on this property), and specify the child node just created as the `target`.
-- This child node becomes the target socket, any external node can be put under this child node and will follow the transformation of the specified skeleton.
+If you need to attach some external nodes to a given skeletal joint, making them transform with skeletal joint together, you may need to use the **Socket System** of the skeletal animation component. Here is a simple example.
+
+### Implementing via the editor
+1. Create a new child node under the skeletal animation component to be docked (the immediate parent node should be the node where the animation component is located).
+2. Add an array element to the `sockets` property of the skeletal animation component, select the `path` of the skeleton to be attached from the drop-down list (note that the defaultClip of the skeletal animation component must have a value, the options in the drop-down list depend on this property), and specify the child node just created as the `target`.
+3. This child node becomes the target socket, any external node can be put under this child node and will follow the transformation of the specified skeleton.
+
+![attach0](./animation/sockets-attach0.png)
+
+### Implementing via code
+```
+    let target = new Node();
+    this.cubeNode.parent = target; // the cubeNode contains a cube model
+    let skeletalAnimation = this.node.getComponent(SkeletalAnimation);
+    target.parent = skeletalAnimation.node; //setting target's parent node which contain a SkeletalAnimation component.
+    let path = "root/_rootJoint/b_Root_00/b_Hip_01/b_Tail01_012/b_Tail02_013/b_Tail03_014";
+    let socket = new SkeletalAnimation.Socket(path, target); // create Socket object with path and target
+    skeletalAnimation.sockets.push(socket);
+```
```

Comment:
```suggestion
```ts
let target = new Node();
this.cubeNode.parent = target; // the cubeNode contains a cube model
let skeletalAnimation = this.node.getComponent(SkeletalAnimation);
target.parent = skeletalAnimation.node; //setting target's parent node which contain a SkeletalAnimation component.
let path = "root/_rootJoint/b_Root_00/b_Hip_01/b_Tail01_012/b_Tail02_013/b_Tail03_014";
let socket = new SkeletalAnimation.Socket(path, target); // create Socket object with path and target
skeletalAnimation.sockets.push(socket);
```
```